pkg/registry/service/rest_test.go

```diff
@@ -461,6 +461,14 @@ func TestServiceRegistryResourceLocation(t *testing.T) {
 		ObjectMeta: api.ObjectMeta{Name: "foo"},
 		Spec: api.ServiceSpec{
 			Selector: map[string]string{"bar": "baz"},
+			Ports: []api.ServicePort{
+				// Service port 9393 should route to endpoint port "p", which is port 93
+				{Name: "p", Port: 9393, TargetPort: util.NewIntOrStringFromString("p")},
+
+				// Service port 93 should route to unnamed endpoint port, which is port 80
+				// This is to test that the service port definition is used when determining resource location
+				{Name: "", Port: 93, TargetPort: util.NewIntOrStringFromInt(80)},
```

Comment:
They do… this wouldn't pass validation, but our fake registry lets us use it. I can split out a separate test case in order to use a completely valid service spec